File: tests/fixtures/sample.dart
class Point {
  final double x;
  final double y;

  Point(this.x, this.y);

  double magnitude() {
    return (x * x + y * y);
  }
}

mixin Logger {
  void log(String msg) {
    print(msg);
  }
}

enum Direction { up, down, left, right }

typedef Pair = (int, int);

int add(int a, int b) {
  return a + b;
}

String greet(String name) {
  return 'Hello, $name!';
}

File: tests/fixtures/sample.kt
package com.example

data class Point(val x: Int, val y: Int)

enum class Direction {
    UP, DOWN, LEFT, RIGHT
}

interface Renderable {
    fun render(): String
}

class Calculator(private var value: Int = 0) {
    fun add(x: Int): Int {
        value += x
        return value
    }

    private fun reset() {
        value = 0
    }
}

object Logger {
    fun log(msg: String) {
        println(msg)
    }
}

typealias Pair = Pair<Int, Int>

fun greet(name: String): String {
    return "Hello, $name!"
}

fun add(a: Int, b: Int): Int {
    return a + b
}
